### PR TITLE
Add pruning to PyTorch example

### PR DESCRIPTION
**Pull request by Crissman**
Our PyTorch example is one of our most popular, but does not show how to use pruning with PyTorch, which is an advantage of Optuna over other hyperparameter optimization libraries. I've added pruning and would like to link to this version for my blog post on PyTorch with Optuna.

### CODE REVIEW

**Comment by Crissman** [comment]
Fixed conflict.

I've made a separate issue for the categorization for pruning examples. #1238 

Please process this PR separately from that issue.

Thanks!

File: examples/pytorch_simple.py [hunk 1]
```diff
@@ -98,41 +98,50 @@ def objective(trial):
...
+        model.eval()
+        correct = 0
+        with torch.no_grad():
+            for batch_idx, (data, target) in enumerate(valid_loader):
+                # Limiting validation data.
+                if batch_idx * BATCHSIZE >= N_VALID_EXAMPLES:
+                    break
+                data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
+                output = model(data)
+                # get the index of the max log-probability
```
Comment (hvy) [inline examples/pytorch_simple.py:119]: A nit, for style and consistency.
```suggestion
                # Get the index of the max log-probability.
```

**Comment by hvy** [comment]
@keisuke-umezawa Sorry if it seems rushed but let me merge this PR and let's continue the discussion about the example structure in the issue mentioned above 🙇
